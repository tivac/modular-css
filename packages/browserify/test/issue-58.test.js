"use strict";

const from = require("from2-string");
const shell = require("shelljs");
const browserify = require("browserify");
const watchify  = require("watchify");
    
const read = require("@modular-css/test-utils/read.js")(__dirname);
const write = require("@modular-css/test-utils/write.js")(__dirname);

const bundle = require("./lib/bundle.js");
const plugin = require("../browserify.js");

describe("/browserify.js", () => {
    describe("/issues", () => {
        describe("/58", () => {
            afterAll(() => shell.rm("-rf", "./packages/browserify/test/output/issues"));
            
            it("should update when CSS dependencies change", (done) => {
                const build = browserify();
                
                write("./issues/58/issue.css", `
                    .issue1 {
                        composes: other1 from "./other.css";
                        color: teal;
                    }

                    .issue2 {
                        composes: issue1;
                        composes: other3 from "./other.css";
                        color: aqua;
                    }
                `);

                write("./issues/58/other.css", `
                    .other1 { color: red; }
                    .other2 { color: navy; }
                    .other3 { color: blue; }
                `);
                
                build.add(
                    from("require('./packages/browserify/test/output/issues/58/issue.css');")
                );

                build.plugin(watchify);
                build.plugin(plugin, {
                    css : "./packages/browserify/test/output/issues/58.css",
                });

                build.on("update", () => {
                    bundle(build).then((out) => {
                        expect(out).toMatchSnapshot();
                        expect(read("./issues/58.css")).toMatchSnapshot();
                    
                        build.close();
                        done();
                    });
                });

                bundle(build).then((out) => {
                    expect(out).toMatchSnapshot();
                    
                    write("./issues/58/other.css", `
                        .other1 { color: green; }
                        .other2 { color: yellow; }
                        .other3 { composes: other2; background: white; }
                    `);
                });
            });
        });
    });
});
