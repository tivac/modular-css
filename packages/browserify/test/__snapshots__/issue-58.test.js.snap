// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/browserify.js /issues /58 should update when CSS dependencies change 1`] = `
"(function e(t,n,r){function s(o,u){if(!n[o]){if(!t[o]){var a=typeof require==\\"function\\"&&require;if(!u&&a)return a(o,!0);if(i)return i(o,!0);var f=new Error(\\"Cannot find module '\\"+o+\\"'\\");throw f.code=\\"MODULE_NOT_FOUND\\",f}var l=n[o]={exports:{}};t[o][0].call(l.exports,function(e){var n=t[o][1][e];return s(n?n:e)},l,l.exports,e,t,n,r)}return n[o].exports}var i=typeof require==\\"function\\"&&require;for(var o=0;o<r.length;o++)s(r[o]);return s})({1:[function(require,module,exports){
require('./packages/browserify/test/specimens/issues/58/issue.css');
},{\\"./packages/browserify/test/specimens/issues/58/issue.css\\":2}],2:[function(require,module,exports){
module.exports = {
    \\"issue1\\": \\"mc2652cf16_other1 mc46e4a65d_issue1\\",
    \\"issue2\\": \\"mc2652cf16_other1 mc46e4a65d_issue1 mc2652cf16_other3 mc46e4a65d_issue2\\"
};
},{\\"./packages/browserify/test/specimens/issues/58/other.css\\":3}],3:[function(require,module,exports){
module.exports = {
    \\"other1\\": \\"mc2652cf16_other1\\",
    \\"other2\\": \\"mc2652cf16_other2\\",
    \\"other3\\": \\"mc2652cf16_other3\\"
};
},{}]},{},[1]);
"
`;

exports[`/browserify.js /issues /58 should update when CSS dependencies change 2`] = `
"(function e(t,n,r){function s(o,u){if(!n[o]){if(!t[o]){var a=typeof require==\\"function\\"&&require;if(!u&&a)return a(o,!0);if(i)return i(o,!0);var f=new Error(\\"Cannot find module '\\"+o+\\"'\\");throw f.code=\\"MODULE_NOT_FOUND\\",f}var l=n[o]={exports:{}};t[o][0].call(l.exports,function(e){var n=t[o][1][e];return s(n?n:e)},l,l.exports,e,t,n,r)}return n[o].exports}var i=typeof require==\\"function\\"&&require;for(var o=0;o<r.length;o++)s(r[o]);return s})({1:[function(require,module,exports){
require('./packages/browserify/test/specimens/issues/58/issue.css');
},{\\"./packages/browserify/test/specimens/issues/58/issue.css\\":2}],2:[function(require,module,exports){
module.exports = {
    \\"issue1\\": \\"mc2652cf16_other1 mc46e4a65d_issue1\\",
    \\"issue2\\": \\"mc2652cf16_other1 mc46e4a65d_issue1 mc2652cf16_other2 mc2652cf16_other3 mc46e4a65d_issue2\\"
};
},{}]},{},[1]);
"
`;

exports[`/browserify.js /issues /58 should update when CSS dependencies change 3`] = `
"/* packages/browserify/test/specimens/issues/58/other.css */
.mc2652cf16_other1 { color: green; }
.mc2652cf16_other2 { color: yellow; }
.mc2652cf16_other3 { background: white; }

/* packages/browserify/test/specimens/issues/58/issue.css */
.mc46e4a65d_issue1 {
    color: teal;
}
.mc46e4a65d_issue2 {
    color: aqua;
}
"
`;
